### PR TITLE
Change cdn to npm package POC

### PR DESCRIPTION
**Pull request by YayunHuang**
## Blocked by 
- #110 

## What's in this PR?
Ref: MUP-161

change cdns in `download-python-package.js`:
- jszip
- file-saver
- localforage(this one also use in `upload.js` so didn't remove cdn file)

### CODE REVIEW

**Comment by pkong-ds** [comment]
Please rebase

**Comment by YayunHuang** [comment]
@pkong-ds rebased

File: src/pages/download/index.astro [hunk 1]
```diff
@@ -41,8 +42,17 @@ import "./index.css";
     }
   };
 
+  const handleDownloadButton = () => {
+    const downloadButton = document.querySelector(".download-btn");
+    downloadButton?.addEventListener("click", () => {
+      // @ts-expect-error // Inject deviceId into window
+      generateZIP(window.targetDeviceId);
```
Comment (pkong-ds) [inline src/pages/download/index.astro:49]: Maybe we should re-think how we pass around data in MUP now; using window as a global context seems very fragile.

But that's another issue

File: src/pages/download/_downloadPythonPackage.js [hunk 2]
```diff
@@ -31,14 +35,15 @@ function getJSZipDateWithOffset() {
...
+export async function generateZIP(deviceId) {
   var zip = new JSZip();
   var count = 0;
   const zipFilename = !!deviceId ? `${deviceId}-mockup.zip` : "mockup.zip";
   var images = new Map();
-  dataurlkey = await allStorage();
+  var dataurlkey = await allStorage();
   dataurlkey.forEach(function (value, key) {
-    file = dataURLtoFile(value, key.substring(3, key.length) + ".png");
+    var file = dataURLtoFile(value, key.substring(3, key.length) + ".png");
```
Comment (pkong-ds) [inline src/pages/download/_downloadPythonPackage.js:46]: const/let does not work?

Comment (YayunHuang) [reply src/pages/download/_downloadPythonPackage.js:46]: can work, but other places in this file all using `var` :sweat_smile:, I can update them all at once when changing `.js` to .ts`

File: src/pages/download/index.astro (hunk 1 above)
Comment (pkong-ds) [reply src/pages/download/index.astro:49]: opened https://linear.app/oursky/issue/MUP-175/restructure-how-data-is-passed-around

**Review by pkong-ds — APPROVED** [review]
LGTM

**Pull request merged**